datafusion/expr/src/expr.rs

```diff
@@ -353,6 +348,22 @@ pub struct GetIndexedField {
     pub key: ScalarValue,
 }
 
+/// BETWEEN expression
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct Cast {
+    /// The expression being cast
+    pub expr: Box<Expr>,
+    /// The `DataType` the expression will yield
+    pub data_type: DataType,
+}
+
+impl Cast {
+    /// Create a new Between expression
+    pub fn new(expr: Box<Expr>, data_type: DataType) -> Self {
+        Self { expr, data_type }
+    }
+}
+
 impl GetIndexedField {
```

Comment:
Why the `struct Cast` is in the middle of `struct GetIndexedField` and `impl GetIndexedField`?